src/test/scala/mesosphere/marathon/api/v2/json/V2AppDefinitionTest.scala

```diff
@@ -17,50 +16,58 @@ import play.api.libs.json.{ JsError, Json }
 
 import scala.collection.immutable.Seq
 import scala.concurrent.duration._
+import com.wix.accord._
+import mesosphere.marathon.api.v2.Validation.getAllRuleConstrains
 
 class V2AppDefinitionTest extends MarathonSpec with Matchers {
 
   test("Validation") {
-    def shouldViolate(app: V2AppDefinition, path: String, template: String) = {
-      val violations = ModelValidation.checkAppConstraints(app, PathId.empty)
-      assert(
-        violations.exists { v =>
-          v.getPropertyPath.toString == path && v.getMessageTemplate == template
-        },
-        s"Violations:\n${violations.mkString}"
-      )
+    def shouldViolate(app: V2AppDefinition, path: String, template: String): Unit = {
+      validate(app) match {
+        case Success => fail()
+        case f: Failure =>
+          val violations = getAllRuleConstrains(f)
+          assert(violations.exists { v =>
+            v.property.contains(path) && v.message == template
+          },
+            s"Violations:\n${violations.mkString}"
+          )
+      }
     }
 
-    def shouldNotViolate(app: V2AppDefinition, path: String, template: String) = {
-      val violations = ModelValidation.checkAppConstraints(app, PathId.empty)
-      assert(
-        !violations.exists { v =>
-          v.getPropertyPath.toString == path && v.getMessageTemplate == template
-        },
-        s"Violations:\n${violations.mkString}"
-      )
+    def shouldNotViolate(app: V2AppDefinition, path: String, template: String): Unit = {
+      validate(app) match {
+        case Success =>
+        case f: Failure =>
+          val violations = getAllRuleConstrains(f)
+          assert(!violations.exists { v =>
+            v.property.contains(path) && v.message == template
+          },
+            s"Violations:\n${violations.mkString}"
+          )
+      }
     }
 
     var app = V2AppDefinition(id = "a b".toRootPath)
-    val idError = "path contains invalid characters (allowed: lowercase letters, digits, hyphens, \".\", \"..\")"
+    val idError = "must fully match regular expression '^(([a-z0-9]|[a-z0-9][a-z0-9\\-]*[a-z0-9])\\.)*([a-z0-9]|[a-z0-9][a-z0-9\\-]*[a-z0-9])|(\\.|\\.\\.)$'"
     validateJsonSchema(app, false)
-    shouldViolate(app, "id", idError)
+    shouldViolate(app, "id.path", idError)
```

Comment:
Probably minor - we already talked about that:
the user defines an id -> we give him a response for id.path